google/cloud/bigquery/_pandas_helpers.py

```diff
@@ -24,6 +24,28 @@
     import pandas
 except ImportError:  # pragma: NO COVER
     pandas = None
+else:
+    import numpy
+
+try:
+    from shapely.geometry.base import BaseGeometry as _BaseGeometry
+except ImportError:  # pragma: NO COVER
+    _BaseGeometry = type(None)
+else:
+    if pandas is not None:  # pragma: NO COVER
+
+        def _to_wkb():
+            from shapely.geos import WKBWriter, lgeos
+
+            write = WKBWriter(lgeos).write
+            notnull = pandas.notnull
+
+            def _to_wkb(v):
+                return write(v) if notnull(v) else v
+
+            return _to_wkb
+
+        _to_wkb = _to_wkb()
```

Comment:
>  think this blob of code could use some additional explanation.

Added,

> I'm guessing you're using the inner function to avoid having to construct a WKBWriter every time `_to_wkb` is called, right? Let's clarify that.

The main reason was to add the not-null check, but then some optimizations beckoned. :)

See the comment I added.

Reply:
> Aside: I'm wondering if we should create an `_init_module()` function or something for this kind of setup code that only runs if all of our expected dependencies are installed? I suppose we might need more than one since `shapely` could be absent even if `pandas` and `pyarrow` are present.

What would be the advantage of adding an `_init_module()` function?